pkg/api/security/platform_nowindows.go

```diff
@@ -9,16 +9,9 @@ package security
 
 import (
 	"io/ioutil"
-	"os"
-
-	"github.com/DataDog/datadog-agent/pkg/config"
 )
 
-// writes auth token(s) to a file with the same permissions as datadog.yaml
+// writes auth token(s) to a file that is only readable by the user running the agent
 func saveAuthToken(token, tokenPath string) error {
-	confFile, err := os.Stat(config.Datadog.ConfigFileUsed())
-	if err != nil {
-		return err
-	}
-	return ioutil.WriteFile(tokenPath, []byte(token), confFile.Mode())
+	return ioutil.WriteFile(tokenPath, []byte(token), 0600)
```

Comment:
would it make sense to do the same thing on Windows? (not sure how permissions work on that file on windows)

Reply:
Windows is already handled correctly : https://github.com/DataDog/datadog-agent/blob/eda9fe5fe0fa5591bc80e203a600f93b08ff61db/pkg/api/security/platform_windows.go#L34-L45